### PR TITLE
Audit log entries for Snapshot Update

### LINKED ISSUE #4283: Add audit log entries for snapshot edit events

### PR DESCRIPTION
**Pull request by Steve-Mcl**
closes #4283

## Description

Adds audit logging entries for Snapshot Editing

### Unit Tests added
```
  Audit Log > Application
    ✔ Provides a logger for application device snapshot updated

  Audit Log > Project
    ✔ Provides a logger for updating snapshots of a project
```

### Screenshots

image

image

## Related Issue(s)

Owner #4283
Parent #3818

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Review by joepavitt — APPROVED** [review]

**Comment by joepavitt** [comment]
Good to go, but will hold off on merge until after I've reviewed/merged https://github.com/FlowFuse/flowfuse/pull/4282

**Comment by Steve-Mcl** [comment]
> Good to go, but will hold off on merge until after I've reviewed/merged #4282

Yeah,  https://github.com/FlowFuse/flowfuse/pull/4282 will be easier to review without this merged in. 

We can merge in any order once both are approved.

**Pull request merged**
